workspaces/ui-v2/src/store/selectors/endpointSelectors.ts

```diff
@@ -0,0 +1,49 @@
+import { IEndpoint, IEndpointWithChanges } from '<src>/types';
+import {
+  EndpointChangelog,
+  ChangelogCategory,
+} from '<src>/hooks/useEndpointsChangelog';
+import { getEndpointId } from '<src>/utils';
+
+export const filterRemovedEndpoints = (endpoints: IEndpoint[]): IEndpoint[] =>
+  endpoints.filter((endpoint) => !endpoint.isRemoved);
+
+/**
+ * Filters removed endpoints not removed in changes and joins changelog information
+ */
+export const filterRemovedEndpointsForChangelogAndMapChanges = (
```

Comment:
This is the main piece of work on the FE (the name is a bit of a mouthful, could split up the functions or rename - but I think it's easier to keep it in one function for it's intended use in changelog):
- An endpoint should show up if it is not removed
- An endpoint that _is_ removed should not show up with no changes (changelog pages should use this page)
- An endpoint that is removed and that deletion is in the range of batch commits _should_ show up with a `changes: 'removed'`
- An endpoint that is removed and that deletion is outside the range of batch commits _should not_ show up